### ISSUE TITLE
Support ZSTD compression in rasters

### ISSUE THREAD

**Issue by latot** [qgis/QGIS#51244]
### Feature description

Hi, actually GDAL has several ways to compress rasters:

https://gdal.org/drivers/raster/gtiff.html

I testes save rasters with ZSTD compression, sadly we can't open that rasters with QGIS, I don't know if is supported or not other compression ways.

QGIS 3.26.3
Gentoo 64
GDAL 3.5.3

Thx!

**Comment by agiudiceandrea** [comment]
@latot could you please provide a sample raster layer? Could you please test using a currently supported version of QGIS (3.28.1 - 3.22.13) with a current version of GDAL (3.6.0). If they are not available for you system, you can tray the [Flatpak](https://flathub.org/apps/details/org.qgis.qgis) versions.

**Comment by nicogodet** [comment]
ZSTD compression read/write works on Windows with OSGeo4W so this feature request is not a FR.
Could come from GDAL used on your system.
"ZSTD is available since GDAL 2.3 when using internal libtiff and if GDAL built against libzstd >=1.0, or if built against external libtiff with zstd support."

**Comment by latot** [comment]
Its weird, I probably can test this the next monday, but the raster I tested was generated with GDAL (coded in python3), explicitly with ZSTD compression......

**Comment by latot** [comment]
Oks, system update today, now with QGIS 3.26.3 and GDAL 3.6.1, in any case is the same as before, the next thing works:

This works
```
import numpy as np
import rasterio
a = rasterio.open("test.tiff", "w", compress="DEFLATE", height = 10, width = 10, count = 1, dtype = "uint8")
a.write(np.ones((10, 10)), 1)
a.close()
```

This does not works:
```
import numpy as np
import rasterio
a = rasterio.open("test.tiff", "w", compress="ZSTD", height = 10, width = 10, count = 1, dtype = "uint8")
a.write(np.ones((10, 10)), 1)
a.close()
```

D:

I can correcyly generate the files, but when we open them with QGIS, the ZSTD one does not works.

Thx!

**Comment by agiudiceandrea** [comment]
@latot I've created the raster layer using the provided Python scripts with QGIS 3.28.1 on Windows (GDAL 3.6.0). They are correctly imported in QGIS.
[test.zip](https://github.com/qgis/QGIS/files/10263262/test.zip)

**Comment by agiudiceandrea** [comment]
@latot I've also tested the DEFLATE and ZSTD compressed GeoTIFF raster layers using QGIS 3.28.2 on Ubuntu 22.04 and they are both correctly imported without issues.
This confirms the issue is due to a packaging issue in the GDAL built in your Gentoo system (and in Flatpak). I suggest you to contact the QGIS/GDAL packagers for the Gentoo system.

**Comment by latot** [comment]
Oks, this is weird!

rasterio seems to work over GDAL (the compression params is a GDAL param, so is a deduction). But after checking gdalinfo:

```
Warning 1: test.tiff: test.tiff:ZSTD compression support is not configured
ERROR 1: test.tiff: Cannot open TIFF file due to missing codec ZSTD.
gdalinfo failed - unable to open 'test.tiff'.
```

So, weird that rasterio can use zstd, well I found other thing, from GDAL:
```
ZSTD is available since GDAL 2.3 when using internal libtiff and if GDAL built against libzstd >=1.0, or if built against external libtiff with zstd support.
```
So, my GDAL does have TIFF, is enabled by default (in gentoo you can choose what libs you use), 
I checked the TIFF lib, and has ZSTD disabled D:
After enable it and recompile tiff now:
```
Driver: GTiff/GeoTIFF
Files: test.tiff
Size is 10, 10
Image Structure Metadata:
  COMPRESSION=ZSTD
  INTERLEAVE=BAND
Corner Coordinates:
Upper Left  (    0.0,    0.0)
Lower Left  (    0.0,   10.0)
Upper Right (   10.0,    0.0)
Lower Right (   10.0,   10.0)
Center      (    5.0,    5.0)
Band 1 Block=10x10 Type=Byte, ColorInterp=Gray
```
Great! now works.

And, thx to this, we know the flatpak version need a fix too.

Thx!

**Comment by agiudiceandrea** [comment]
Issue report submitted for QGIS packaged by Flatpak https://github.com/flathub/org.qgis.qgis/issues/254.
